### PR TITLE
Camera and gallery now display after granting permission

### LINKED ISSUE #64: Camera doesn't work in Android
**Describe the bug**
Starting a new observation by taking a photo does not work in Android. I just get a black screen of death. Furthermore the app doesn't seem to ask for the camera permission (black screen happens regardless).

**To Reproduce**
1. Sign in, tap "camera" in the tab bar, and tap "take photo with camera"
2. You'll see an alert asking you to allow the app to "access photos and media on your device"; tap "Allow"
3. The screen is black except for the OS UI at top and bottom. You can tap the system back button to get back to the app, but there's no camera UI
4. Go to the OS settings and find iNaturalistReactNative under Apps
5. Tap "Permissions" and you'll see that the Camera permission hasn't been allowed; tap it to explicitly allow it
6. Return to the app and try to take a photo again; you still get the black screen even though the app has the camera permission

**Expected behavior**
I'd like to take a photo, please.

**Screenshots**

**Context (please complete the following information):**
 - Device: Pixel 3
 - OS: Android 12
 - App version: 0.0.1

### LINKED ISSUE #65: Can't view gallery photos after granting permission in Android
**Describe the bug**
After granting the file access permission for the first time, the photos never load until you force quit the app.

**To Reproduce**
1. In a fresh install of the app, sign in, tap "camera" in the tab bar, and tap "upload photo from gallery"
2. You should see an alert asking for access to photos and files; tap "Allow"
3. You land on a screen that has the text "camera roll" and a spinner that spins forever
4. Force quit the app and repeat the above steps; your recent photos appear as you'd expect

**Expected behavior**
After you grant the permission, you should see the photos you just granted the app permission to access.

**Context (please complete the following information):**
 - Device: Pixel 3
 - OS: Android 12
 - App version: 0.0.1

**Additional context**
FWIW, I strongly recommend not using custom UI for photo selection and using an `ACTION_GET_CONTENT` intent [like we do in the native app](https://github.com/inaturalist/iNaturalistAndroid/blob/main/iNaturalist/src/main/java/org/inaturalist/android/ObservationEditor.java#L1497-L1513). Using the intent-based approach we use in the native Android app allows users to browse photos from different apps, navigate their local filesystem, search for files, sort files, view files before choosing, etc.

### PR DESCRIPTION
**Pull request by kueda**
Previously these were not quite working because the usePhotos hook was asking for a permission, so if that was used anywhere in conjunction with another permission request, one or both success promises never got executed. This commit tries to solve that with a PermissionGate component that takes a permission as a prop, asks for it if not granted, then renders its children when granted. If not granted, it renders a message saying that the permission is required to use the feature.

Other stuff

* Added `env.example` so people know how to populate `.env`
* Added the `intl-pluralrules` polyfill to support pluralization with i18next
* Skip some token granting logic if the user is not logged in
* Trim whitespace from username/email before auth
* Some minor UI changes to auth form so I know where to tap
* Test fixes

Closes #64
Closes #65

### CODE REVIEW

File: src/styles/login/login.js [hunk 1]
```diff
@@ -22,7 +22,10 @@ const textStyles: { [string]: TextStyleProp } = StyleSheet.create( {
   },
   input: {
     color: "#000000",
-    backgroundColor: "#ffffff"
+    backgroundColor: "#ffffff",
```
Comment (albullington) [inline src/styles/login/login.js:25]: Pondering of the week and something I wish I had thought of earlier: should we be using a component library like [react-native-elements](https://reactnativeelements.com/) for placeholders while we're waiting for final designs? That might make it easier to navigate the app, and ideally we can find a component library that will allow us to customize styling.

File: src/navigation/cameraStackNavigation.js [hunk 2]
```diff
@@ -2,28 +2,46 @@
...
+import PhotoGallery from "../components/PhotoLibrary/PhotoGallery";
+import PermissionGate from "../components/SharedComponents/PermissionGate";
 
 const Stack = createNativeStackNavigator( );
 
 const hideHeader = {
   headerShown: false
 };
 
+const PhotoGalleryWithPermission = ( ) => (
```
Comment (albullington) [inline src/navigation/cameraStackNavigation.js:23]: What's your thinking behind using `PermissionGate` for the camera and photo gallery, but not the sound recorder? Wouldn't it be easiest to keep all Android permissions code in the same location?

**Review by albullington — COMMENTED** [review]
Looks good to me and definitely works a lot better than it did before. I didn't realize hooks / reusable components would negatively impact permissions, so glad to learn something new here.

Left a few questions / nitpicks.

File: src/components/UserProfile/UserProfile.js [hunk 3]
```diff
@@ -70,7 +70,7 @@ const UserProfile = ( ): React.Node => {
       <Text>Bio:</Text>
       <HTML
         contentWidth={width}
-        source={{ html: user.description }}
+        html={user.description}
```
[the remark this answers is not in the document]
Comment (kueda) [reply src/components/UserProfile/UserProfile.js:73]: It works. I think I changed it b/c it wasn't working for me with https://www.npmjs.com/package/react-native-render-html/v/6.3.4, which seems to be what we're specifying in package.json.

File: src/styles/login/login.js (hunk 1 above)
Comment (kueda) [reply src/styles/login/login.js:25]: Yes please. No matter what the final design is, a theming system is a good idea for consistency.

File: src/navigation/cameraStackNavigation.js (hunk 2 above)
Comment (kueda) [reply src/navigation/cameraStackNavigation.js:23]: My thinking was just that I was trying to address the two issues I filed, but yes, if you're cool with this approach, I think we should apply it elsewhere after merging.

File: src/components/SharedComponents/PermissionGate.js [hunk 4]
```diff
@@ -0,0 +1,98 @@
...
+      ) }
+    </ViewNoFooter>
+  );
+
+  let content = <Text>Requesting permission...</Text>;
+  if ( result === "granted" ) { content = children; }
+  else if ( result !== null ) { content = noPermission; }
+
+  return (
+    <ViewNoFooter>
```
[the remark this answers is not in the document]
Comment (kueda) [reply src/components/SharedComponents/PermissionGate.js:92]: Fine w/ me, but the camera seems to be a bit broken on Android so I can't quite test how this looks on the iPhone. Do you mean returning something like this?

```js
<View style={{ flex: 1 }}>
  { content }
</View>
```

[the remark this answers is not in the document]
Comment (albullington) [reply src/components/SharedComponents/PermissionGate.js:92]: Yep, exactly

File: src/components/UserProfile/UserProfile.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (albullington) [reply src/components/UserProfile/UserProfile.js:73]: Interesting. It's not working for me with 6.3.4, but I guess we can always change later.

File: src/navigation/cameraStackNavigation.js (hunk 2 above)
Comment (albullington) [reply src/navigation/cameraStackNavigation.js:23]: Gotcha, sounds like a plan!

File: src/components/UserProfile/UserProfile.js (hunk 3 above)
[the remark this answers is not in the document]
Comment (kueda) [reply src/components/UserProfile/UserProfile.js:73]: What's your package-lock.json look like? Mine:

```
    "node_modules/react-native-render-html": {
      "version": "6.3.4",
      "resolved": "https://registry.npmjs.org/react-native-render-html/-/react-native-render-html-6.3.4.tgz",
      "integrity": "sha512-H2jSMzZjidE+Wo3qCWPUMU1nm98Vs2SGCvQCz/i6xf0P3Y9uVtG/b0sDbG/cYFir2mSYBYCIlS1Dv0WC1LjYig==",
      "dependencies": {
        "@jsamr/counter-style": "^2.0.1",
        "@jsamr/react-native-li": "^2.3.0",
        "@native-html/transient-render-engine": "11.2.3",
        "@types/ramda": "^0.27.40",
        "@types/urijs": "^1.19.15",
        "prop-types": "^15.5.7",
        "ramda": "^0.27.2",
        "stringify-entities": "^3.1.0",
        "urijs": "^1.19.6"
      },
      "peerDependencies": {
        "react": "*",
        "react-native": "*"
      }
    },
```

[the remark this answers is not in the document]
Comment (albullington) [reply src/components/UserProfile/UserProfile.js:73]: Looks the same to me...

```
 "node_modules/react-native-render-html": {
      "version": "6.3.4",
      "resolved": "https://registry.npmjs.org/react-native-render-html/-/react-native-render-html-6.3.4.tgz",
      "integrity": "sha512-H2jSMzZjidE+Wo3qCWPUMU1nm98Vs2SGCvQCz/i6xf0P3Y9uVtG/b0sDbG/cYFir2mSYBYCIlS1Dv0WC1LjYig==",
      "dependencies": {
        "@jsamr/counter-style": "^2.0.1",
        "@jsamr/react-native-li": "^2.3.0",
        "@native-html/transient-render-engine": "11.2.3",
        "@types/ramda": "^0.27.40",
        "@types/urijs": "^1.19.15",
        "prop-types": "^15.5.7",
        "ramda": "^0.27.2",
        "stringify-entities": "^3.1.0",
        "urijs": "^1.19.6"
      },
      "peerDependencies": {
        "react": "*",
        "react-native": "*"
      }
    },
```

**Pull request merged**